rfc/2022/2022-06-16_platform-leadership_002_github-access-managed-nested-teams.md

```diff
@@ -0,0 +1,126 @@
+# RFC: Github Access for Platform and VFS users will be managed by nested Github Teams
+
+<!--
+The title is what you want comments on. Use the active voice in a future tense.
+Example:
+    - The website will be built using the XZY framework
+-->
+
+* Comment Deadline: `YYYY-MM-DD`
+* Team Crew and Name: Platform/Leadership
+* Authors:
+  * @mchelen-gov
+  * [Contributing Author 1 Github Handle](https://github.com/ContributingAuthor1Handle)
+  * [Contributing Author 2 Github Handle](https://github.com/ContributingAuthor2Handle)
+* [Original RFC Pull Request](https://github.com/department-of-veterans-affairs/va.gov-platform-arch/pull/-1)
+
+## Background
+
+
+Currently permissions to repos such as va.gov-team, vets-website, and vets-api are largely handled through the `vets.gov-write` [team](https://github.com/orgs/department-of-veterans-affairs/teams/vets-gov-write). This team includes a mixture of individual [members](https://github.com/orgs/department-of-veterans-affairs/teams/vets-gov-write/members) as well as [child teams](https://github.com/orgs/department-of-veterans-affairs/teams/vets-gov-write/teams).
+
+This makes management of users and permissions difficult for multiple reasons:
+1. The roles of various individual users added to the `vets.gov-write` team are unclear and difficult to track.
+1. Onboarding and offboarding of individual users has to be performed by maintainers of `vets.gov-write` creating maintainence burden.
+1. Refined permissions cannot be applied to various level groupings such as "all Platform teams" or "all VFS teams"
+
+We need to restructure our Github Teams to improve clarity, reduce manual overhead, and increase granularity of permissions.
+
+
+## Motivation
+
+Which problem(s) does this solve? Why? Use active voice, present tense, and decisive language.
+Example: Framework XYZ is a modern, well-known, web rendering framework. It is easy to learn, easy to use, and provides high-performance web rendering. etc
+
+## Design
+
+Explain the proposed design in enough detail so that a team member will fully understand the implementation. Include a diagram (in the `images` dir) as needed to convey your plans. Use active voice, present tense, and decisive language.
+
+### Phase 1 - Establish structure
+
+First, use the existing `vets.gov-write` team and ensure all users are represented in a child team.
+
+```mermaid
+graph TD
+    subgraph vets.gov-write
```

Comment:
Jumping in to comment here since we have some relevant functionality regarding visualizing teams. Take a look at the Console UI depiction of vets.gov-write and see how it might be used to help visualize the organization of teams.

https://platform-console-ui-utility.vfs.va.gov/catalog/default/group/vets-gov-write

![Screen Shot 2022-07-26 at 11 16 15 AM](https://user-images.githubusercontent.com/116142/181044324-dcc09f62-1b61-444e-b071-10f53c60d7eb.png)

The upper left card shows the teams that are children of the current team.

Ownership of projects is in the middle card.

We added a Links card for links to Confluence, Issue Templates, important team policies.

Open pull requests shows when PR reviews have been requested of a team...

Membership of a team follows further down the page.

![Screen Shot 2022-07-26 at 11 19 01 AM](https://user-images.githubusercontent.com/116142/181044996-6e717564-5ea0-499d-9be1-a360a95940ce.png)

There's nothing that shows the permissions of the team or its members, but we might be able to depict that by writing a plugin that leverages the GitHub API.

Reply:
https://platform-console-ui-utility.vfs.va.gov/catalog-graph?rootEntityRefs%5B%5D=group%3Adefault%2Fplatform-pst-console-ui&maxDepth=4&selectedKinds%5B%5D=group&selectedRelations%5B%5D=parentOf&selectedRelations%5B%5D=childOf&unidirectional=true&mergeRelations=true&direction=LR&showFilters=true

![Screen Shot 2022-07-29 at 3 20 23 PM](https://user-images.githubusercontent.com/116142/181829750-74ee213f-3078-40db-809d-70d3a1aa1c58.png)

Also, Console UI can graph team relationships.